src/Lifecycle.ts

```diff
@@ -602,7 +607,12 @@ async function doSetLoggedIn(credentials: IMatrixClientCreds, clearStorageEnable
     //
     // we fire it *synchronously* to make sure it fires before on_logged_in.
     // (dis.dispatch uses `window.setTimeout`, which does not guarantee ordering.)
-    dis.dispatch({ action: "on_logging_in" }, true);
+    //
+    // can be disabled to resolve "Cannot dispatch in the middle of a dispatch."
+    // error when it is invoked via another dispatch that is not yet finished.
+    if (dispatchOnLoggingIn) {
+        dis.dispatch({ action: Action.OnLoggingIn }, true);
+    }
```

Comment:
Are we sure this is actually still needed? If I'm not mislead, the `on_loggin_in` action was removed back in 2020 [in this commit](https://github.com/matrix-org/matrix-react-sdk/commit/27ee90cad58666a97e3a61993f3a7c2575753aa2#diff-0a70cc9391fac2588db9133b2344089e9e6ae6f215fd3f36c81fc985892c762bL659). There is also no other reference in the current code that uses it, so removing it might reduce the complexity of this PR.

Reply:
removing this code sounds good to me, @justjanne what do you think?

Reply:
Just checked with a colleague, sounds good :+1:

Reply:
This code is removed. @justjanne could you please review?